### PR TITLE
update: enhance zod schema validation for Login and Register pages

### PR DESCRIPTION
**Pull request by Maaayhan**
Resolved CP-126

[CP-126](https://beequant.atlassian.net/jira/software/projects/CP/boards/1?selectedIssue=CP-126)

---

## What happens here?
- update schema for the login page to ensure 'email' conforms to the standard and 'password' is not empty.
- update the schema for the registration page to ensure all fields conform to the standards and 'confirmPassword' matches 'password'.
- add multilingual support.
- add passwordSchema in 'utils' for reuse.

## How do I know this is working?
- when the user enters an invalid email address or leaves the password field empty on login page, a prompt message will appear.
image

- when the user inputs data that does not meet the standards on register page, a message will be displayed
- if the confirm password does not match the password, the message will be shown
image

### CODE REVIEW

File: apps/client/src/module/auth/login-form.tsx [hunk 1]
```diff
@@ -15,12 +15,21 @@ import { SocialButton } from "../common/social-button";
...
 
 type LoginForm = z.infer<typeof formSchema>;
 // source of truth for this login form <= this is the law, is the king, is the absolute authority for this form - K总
 const formSchema = z.object({
-  email: z.string().email(),
-  password: z.string().min(8),
+  email: z.string()
+    .min(1, { message: EmailErrorMsgs.Required })
+    .email({ message: EmailErrorMsgs.Invalid })
+    .regex(emailPattern, { message: EmailErrorMsgs.Invalid }),
```
Comment (NorrisWu0) [inline apps/client/src/module/auth/login-form.tsx]: What's special about this email pattern?

File: apps/client/src/module/auth/register-form.tsx [hunk 2]
```diff
@@ -12,16 +12,51 @@ import { AuthFormContainer } from "./auth-form-container";
...
+    .max(32, { message: PasswordErrorMsgs.MaxLength })
+    .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
+    confirmPassword: z.string()
+    .min(1, { message: ConfirmPasswordErrorMsgs.Required })
+    .min(8, { message: PasswordErrorMsgs.MinLength })
+    .max(32, { message: PasswordErrorMsgs.MaxLength })
+    .regex(passwordPatten, { message: PasswordErrorMsgs.Invalid }),
   realName: z.string().optional(),
-  ref: z.string(),
+  ref: z.string().min(1, { message: RefErrorMsgs.Required }),
```
Comment (NorrisWu0) [inline apps/client/src/module/auth/register-form.tsx]: Oh. Do we have a default value for this?

File: apps/client/src/module/auth/login-form.tsx (hunk 1 above)
Comment (Maaayhan) [reply apps/client/src/module/auth/login-form.tsx]: The email pattern is left when joi was used before, has the same effect as.email(). I'll delete that line

File: apps/client/src/module/auth/register-form.tsx (hunk 2 above)
Comment (Maaayhan) [reply apps/client/src/module/auth/register-form.tsx]: yeah we have a pre-filled default value corresponding to the backend, I‘ll add a default value specification here

**Comment by Maaayhan** [comment]
For the schema of displayName in register page, i found that if user input an empty string, the operation of converting to the default value "New User" was completed in the schema of the back-end, so I wrote the front-end part as: meet the format requirements of display or the length of string was zero.

File: apps/client/src/module/auth/login-form.tsx [hunk 3]
```diff
@@ -17,23 +17,28 @@ import { login, LoginPayload } from "./auth-service";
...
+    password: z.string()
+      .min(1, { message: t("Notifications.password.required") }),
+    remeber: z.string(),
+  });
+
+  // Need this to enforce default value to match the LoginForm schema, because if you do it in the `useForm` hook, it turns it into a `Partial<LoginForm>` type.
+  const defaultValues = {
+    email: "",
+    password: "",
+    remeber: "",
```
Comment (NorrisWu0) [inline apps/client/src/module/auth/login-form.tsx]: oops, looks like a typo here

**Review by NorrisWu0 — DISMISSED** [review]
lgtm, except that typo. feel free to merge this after that's fixed 🥳

**Review by NorrisWu0 — APPROVED** [review]

**Pull request merged**